src/scrape_up/finance/bse.py

```diff
@@ -0,0 +1,111 @@
+    fetcher.headers.update(headers)
+
+    def __init__(self, stock_name):
+        self.stock_name = stock_name
+        self.stock_code = self.get_data()["code"]
+        self.latest_price_url = self.latest_price_url.format(code=self.stock_code)
+        self.chart_data_url = self.chart_data_url.format(code=self.stock_code)
+
+    # gets closest matching name and symbol of stock based given stock_name
+    def get_data(self):
+        try:
+            data_html = self.fetcher.get(
+                self.autocomplete_url.format(self.stock_name)
+            )
+            soup = BeautifulSoup(data_html.text, "html.parser")
+            li_element = soup.find("li", class_="quotemenu")
+            stock_code = li_element.find("span").text.split()[-1]
+            stock_name = li_element.find("strong").text
+            if(li_element.find("strong").next_sibling.strip):
+                stock_name += str(li_element.find("strong").next_sibling.strip())
+
+        except IndexError:
+            raise Exception("Invalid Stock Name.")
+        except TypeError:
+            raise Exception("Invalid Stock Name")
+        except requests.exceptions.ConnectionError:
+            raise Exception("Connection Error, Please try again.")
+        else:
+            self.stock_name = stock_name
+            return {
+                "code": stock_code,
+                "symbol_info": stock_name
+                }
+
+    def get_latest_price(self):
+        """
+        Gets Latest stock price info of given bse stock.
+        """
+        price_info = self.fetcher.get(
+            self.latest_price_url
+            ).json()["CurrRate"]
+        return {
+            "latestPrice": float(price_info["LTP"]),
+            "rateChange": float(price_info["Chg"]),
+            "pChange": float(price_info["PcChg"])
+        }
+
+    # Please give dates in DD-MM-YYYY format
+    def get_historical_data(self, from_date, to_date):
+        """
+        Gets historical stock price (vwap) in range from_date to to_date
+        """
+
+        from_date = datetime.strptime(from_date, "%d-%m-%Y").strftime("%Y%m%d")
+        to_date = datetime.strptime(to_date, "%d-%m-%Y").strftime("%Y%m%d")
+        
+        historical_price_data_raw = self.fetcher.get(
+            self.historical_data_url.format(
+                code=self.stock_code, from_date=from_date, to_date=to_date
+            )
+        ).json()["Data"]
+
+        historical_price_data_raw = json.loads(historical_price_data_raw)
+        historical_price_data = {}
+
+        for item in historical_price_data_raw:
+            dttm = datetime.strptime(item["dttm"], "%a %b %d %Y %H:%M:%S").strftime("%d-%m-%Y")
+            vale1 = float(item["vale1"])
+            historical_price_data[dttm] = vale1
+
+        return historical_price_data
+
+if __name__=="__main__":
+    infosys = BSE("infosys ltd")
+    print(infosys.get_latest_price())
+    print(infosys.get_historical_data("05-06-2023", "10-06-2023"))
+
```

Comment:
Please cross check this one 
![image](https://github.com/Clueless-Community/scrape-up/assets/93156825/a86f7c41-70d0-476c-8041-bc2788be0c80)

Reply:
Hey @nikhil25803 its working fine in in my system:
![Screenshot from 2023-06-11 11-17-52](https://github.com/Clueless-Community/scrape-up/assets/71490015/fbea593e-956d-4509-9f7c-6c54416f973d)

Even StockPrice class is working:
![Screenshot from 2023-06-11 11-20-38](https://github.com/Clueless-Community/scrape-up/assets/71490015/4619319a-a3e1-4700-81fa-16f90a4c985d)

Maybe try updating your modules on which bse.py depends? And also try to look at request content returned wherever the content is being parsed as json. Please try to go through the request output using debug console (please try to avoid modifying code). Please let me know what the result is.